Wrote notebook on synthetic Mosviz image creation

ojustino opened the pull request:
I've opened this PR to get this notebook that I wrote with @robelgeda on the record, but please consider it a draft until we hear more from other stakeholders in the JWST data pipeline discussion. I realize the notebook still contains some paths that are specific to my laptop and that the style may not be fully compliant with `jdaviz` standards.

pllim commented:
As per your comment, I converted this PR to draft status. You can convert it back to "ready" when it is ready. FYI.

robelgeda reviewed (approved):
This is a well written notebook and fulfills the all the requirements requested 🎉

robelgeda commented:
codecov fails but this this a notebook so it may not apply

pllim commented:
FYI: For the future, please open a PR from a branch on your fork, not on this repo.

pllim commented on notebooks/concepts/generate_mosviz_photometry.ipynb:
```diff
@@ -0,0 +1,837 @@
+    "- The objects from `photutils.segmentation` do the work of analytically finding bright sources, separating those that overlap, and creating a catalog of the resulting information.\n",
+    "- We use `matplotlib.pyplot` to preview the field image, the cutouts, and various stages of our synthetic image.\n",
+    "- We use `numpy` to facilitate several specialized mathematical and array-related operations.\n",
+    "- `os` helps us create a new directory for the final cutouts from the synthetic image."
```
```suggestion
    "- `os` helps us create a new directory for the final cutouts from the synthetic image.\n"
    "- `%matplotlib inline` is a notebook magic to show the plot within the notebook."
```

pllim commented on notebooks/concepts/generate_mosviz_photometry.ipynb:
```diff
@@ -0,0 +1,837 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%script false --no-raise-error\n",
```
Instead of inserting `%%script`, why not convert the cell to "raw NBconvert" type instead? That way, when someone wants to run it, they convert it back to actual code cell.

ojustino replied:
Are there issues with `%%script` to be concerned about? I was hoping to preserve those cells' syntax highlighting while also not having them run by default.

pllim replied:
It is fine if you want to keep them, especially since this is an exploratory notebook.

pllim commented on notebooks/concepts/generate_mosviz_photometry.ipynb:
```diff
@@ -0,0 +1,837 @@
+   "source": [
+    "# view level 3 spectra FITS header information\n",
+    "filepath = '/user/jotor/jwst-pipeline-lvl3-copy/'\n",
+    "x1d_header = fits.getheader(filepath + 'jw00626-o030_s00000_nirspec_f170lp-g235m_x1d.fits', ext=1)\n",
+    "#s2d_header = fits.getheader(filepath + 'jw00626-o030_s00000_nirspec_f170lp-g235m_s2d.fits'', ext=1)"
```
If `s2d_header` is not used, just delete this line.

```suggestion
    "x1d_header = fits.getheader(filepath + 'jw00626-o030_s00000_nirspec_f170lp-g235m_x1d.fits', ext=1)"
```

pllim commented on notebooks/concepts/generate_mosviz_photometry.ipynb:
```diff
@@ -0,0 +1,837 @@
+   "source": [
+    "fig, ax = plt.subplots(figsize=(10,10))\n",
+    "ax.imshow(synth_image, vmin=0, vmax=synth_image.std(), **imshow_params)\n",
+    "#ax.imshow(synth_image, vmin=0, vmax=synth_image.mean()*3, **imshow_params)\n",
```
```suggestion
```

pllim reviewed (commented):
Very nice and detailed walkthrough. Thanks!

Branding question: Is it "Mosviz" or "MOSViz" in the narrative doc? Maybe @javerbukh knows. But then again, maybe this is not important to hash out within the "concepts" notebooks.

Disclaimer: I did not run the cells that has `%%script` and require VPN.

Minor comments below.

Also, the paths are not Windows-friendly but you don't have to do anything about it here; just something I noticed.

ojustino commented:
@pllim Thanks for the review. My mistake on opening a pull request from upstream -- I actually already have my fork and remote set up and can open a new PR once the notebook is cleared for merging.

I could work on putting the data on Box in order to drop the VPN requirement. I addressed most of your comments in the new commit and left a reply on the last one.

pllim reviewed (approved):
> I could work on putting the data on Box in order to drop the VPN requirement.

Depends on your audience. If they are all employees, I wouldn't sweat it.
